src/main/java/nextstep/reservation/ReservationService.java

```diff
@@ -21,40 +23,31 @@ public ReservationService(ReservationDao reservationDao, ThemeDao themeDao, Sche
         this.scheduleDao = scheduleDao;
     }
 
-    public Long create(ReservationRequest reservationRequest) {
-        Schedule schedule = scheduleDao.findById(reservationRequest.getScheduleId());
-        if (schedule == null) {
-            throw new NullPointerException();
-        }
-
-        List<Reservation> reservation = reservationDao.findByScheduleId(schedule.getId());
-        if (!reservation.isEmpty()) {
-            throw new DuplicateEntityException();
-        }
-
+    public Long create(Long scheduleId, LoginMember loginMember) {
+        Schedule schedule = scheduleDao.findById(scheduleId).orElseThrow(ScheduleNotFoundException::new);
+        reservationDao.findByScheduleId(schedule.getId())
+                .ifPresent(s -> {throw new DuplicateReservationException();});
         Reservation newReservation = new Reservation(
                 schedule,
-                reservationRequest.getName()
+                loginMember.getUsername()
         );
 
         return reservationDao.save(newReservation);
     }
 
-    public List<Reservation> findAllByThemeIdAndDate(Long themeId, String date) {
-        Theme theme = themeDao.findById(themeId);
-        if (theme == null) {
-            throw new NullPointerException();
-        }
-
-        return reservationDao.findAllByThemeIdAndDate(themeId, date);
+    public List<ReservationResponse> findAllByThemeIdAndDate(Long themeId, String date) {
+        Theme theme = themeDao.findById(themeId).orElseThrow(ThemeNotFoundException::new);
+        return reservationDao.findAllByThemeIdAndDate(themeId, date)
+                .stream()
+                .map(reservation -> ReservationResponse.of(reservation))
```

Comment:
메서드 레퍼런스를 사용해보는건 어떨까요?

```suggestion
                .map(ReservationResponse::of)
```